### PR TITLE
examples: provide keys and certificates for running TLS example, update its README

### PR DESCRIPTION
**Pull request by voidzcy**
The way of generating self-signed keys and certificates in TLS example does not work, since the generated certificates does not contain Subject Alternative Names. 

Generating self-sign certificates (which requires special config for subject alternative names) is quite involved and users may encounter problems doing that (#5737). Therefore, we supply users with our test keys and certificates (I've modified and regenerated the certificates for localhost domain) for them to easily run the example.

### CODE REVIEW

File: examples/example-tls/README.md [hunk 1]
```diff
@@ -19,6 +19,9 @@ in the
 `build/install/example-tls/bin/` directory that run the example. The
 example requires the server to be running before starting the client.
 
+
+## How to run
```
Comment (dapengzhang0) [inline examples/example-tls/README.md]: Level 2 heading not needed. If added, "To build the example" above should also be a heading of the same level.

File: examples/example-tls/README.md [hunk 2]
```diff
@@ -35,65 +38,27 @@ USAGE: HelloWorldClientTls host port trustCertCollectionFilePath [clientCertChai
...
-openssl pkcs8 -topk8 -nocrypt -in server.key -out server.pem
-popd
-```
 
 #### Hello world example with TLS (no mutual auth):
 
 ```bash
 # Run the server:
-./build/install/example-tls/bin/hello-world-tls-server 50440 /tmp/sslcert/server.crt /tmp/sslcert/server.pem
+./build/install/example-tls/bin/hello-world-tls-server 50440 ./src/main/resources/certs/server.pem ./src/main/resources/certs/server.key
```
Comment (dapengzhang0) [inline examples/example-tls/README.md]: You need also update the Maven and Bazel sections below.

File: examples/example-tls/README.md [hunk 3]
```diff
@@ -35,65 +38,27 @@ USAGE: HelloWorldClientTls host port trustCertCollectionFilePath [clientCertChai
   Note: clientCertChainFilePath and clientPrivateKeyFilePath are only needed if mutual auth is desired.
 ```
 
-#### Generating self-signed certificates for use with grpc
-
-You can use the following script to generate self-signed certificates for grpc-java including the hello world with TLS examples:
+We provide a set of self-signed client/server keys and certificates to run the hello world with TLS examples on 
+`localhost`. They can be found in `src/main/resources/certs`. You will need to generate/supply your own 
+client/server keys and certificates with Subject Alternative Names in other domains.
```
Comment (dapengzhang0) [inline examples/example-tls/README.md]: What about "You can provide your own keys and certificates but be aware that the system may require a valid Subject Alternative Names."

File: examples/example-tls/src/main/java/io/grpc/examples/helloworldtls/HelloWorldClientTls.java [hunk 4]
```diff
@@ -123,12 +123,7 @@ public static void main(String[] args) throws Exception {
         }
 
         try {
-            /* Access a service running on the local machine on port 50051 */
-            String user = "world";
-            if (args.length > 0) {
-                user = args[0]; /* Use the arg as the name to greet if provided */
-            }
-            client.greet(user);
+            client.greet(args[0]);
```
Comment (dapengzhang0) [inline examples/example-tls/src/main/java/io/grpc/examples/helloworldtls/HelloWorldClientTls.java]: I would not support the case (args.length == 2) for simplicity.

File: examples/example-tls/README.md (hunk 1 above)
Comment (voidzcy) [reply examples/example-tls/README.md]: Done.

File: examples/example-tls/README.md (hunk 2 above)
Comment (voidzcy) [reply examples/example-tls/README.md]: Done.

File: examples/example-tls/README.md (hunk 3 above)
Comment (voidzcy) [reply examples/example-tls/README.md]: Done.

File: examples/example-tls/src/main/java/io/grpc/examples/helloworldtls/HelloWorldClientTls.java (hunk 4 above)
Comment (voidzcy) [reply examples/example-tls/src/main/java/io/grpc/examples/helloworldtls/HelloWorldClientTls.java]: Done.

**Comment by ejona86** [comment]
I'm not wild about having another set of certs to maintain. But if we are, we absolutely must have configuration/details on how to regenerate them, even if it says "modify that configuration file over there and run the commands mentioned in X".

**Comment by voidzcy** [comment]
@ejona86 

What about just providing the config file and new instructions for how to generate certs that can be used for this example (similar to the instructions in test certs).

**Comment by dapengzhang0** [comment]
I would copy and reuse some of the certs in the testing package, and reference where they are coming from and the README file in the testing package about how they are generated.

**Comment by voidzcy** [comment]
> I would copy and reuse some of the certs in the testing package, and reference where they are coming from and the README file in the testing package about how they are generated.

If we use the test certs directly for the example, we need to add `managedChannelBuilder.overrideAuthority("foo.test.google.fr")` in order to match the subject alternative names in the test certs. I am thinking about providing them [this config file](https://github.com/grpc/grpc-java/blob/master/testing/src/main/resources/certs/server1-openssl.cnf) while have comments on the `alter_names` section. Then we copy and paste [instructions in test certs README](https://github.com/grpc/grpc-java/blob/master/testing/src/main/resources/certs/README) to show how certs are generated.

**Comment by dapengzhang0** [comment]
I'm OK with modifying the config file in the testing package, just make sure all the current interop-testings won't break.

**Comment by ejona86** [comment]
We shouldn't modify the config in the testing package. 1) I think there's another copy internally 2) I don't want localhost in our test certs as it is "wrong".

**Comment by voidzcy** [comment]
Close, please see #5763 .

**Pull request closed without merging**